session/txn.go

```diff
@@ -293,7 +310,11 @@ func (s *session) getTxnFuture(ctx context.Context) *txnFuture {
 
 	oracleStore := s.store.GetOracle()
 	tsFuture := oracleStore.GetTimestampAsync(ctx)
-	return &txnFuture{tsFuture, s.store}
+	ret := &txnFuture{future: tsFuture, store: s.store}
+	if x := ctx.Value("mockGetTSFail"); x != nil {
```

Comment:
Can we use gofail to mock the error? To avoid hurting the performance.

Reply:
No we can't. `gofail` is works globally, 
 `create table t ...` will run some background job and hang in unit test

```
create table t ...
// gofail enable here, the test code hang
select * from t ...     
insert into t values ...
```

Reply:
Could we use gofail + global variable?

Reply:
I advice you using below code:

```
if globalVariableEnable && x := ctx.Value("mockGetTSFail"); x != nil {
```

Reply:
The performance is good in general:
https://golang.org/src/context/context.go#L183 @winkyao

Reply:
Never mind.